colossalai/kernel/cuda_native/fmha/flash_attn_2.py

```diff
@@ -0,0 +1,72 @@
+import warnings
+
+import torch
+
+
+def is_ampere_or_better_gpu():
+    if torch.cuda.is_available():
+        device = torch.device("cuda")
+        properties = torch.cuda.get_device_properties(device)
+        if properties.major >= 8:    # Ampere GPUs or newer
+            return True
+    return False
+
+
+# "Check Ampere GPUs or newer"
+HAS_FLASH_ATTN = False
+if is_ampere_or_better_gpu():
+    HAS_FLASH_ATTN = True
+else:
+    warnings.warn('FlashAttention only supports Ampere GPUs or newer.')
+    HAS_FLASH_ATTN = False
+try:
+    from flash_attn.flash_attn_interface import flash_attn_func, flash_attn_varlen_func
+    HAS_FLASH_ATTN = True
+except ImportError:
+    warnings.warn('please install flash_attn from https://github.com/HazyResearch/flash-attention')
+    HAS_FLASH_ATTN = False
+
+if HAS_FLASH_ATTN:
+    from einops import rearrange
+
+    from ..scaled_softmax import AttnMaskType
+    from .utils import SeqLensInfo
+
+    def flash_attention(q,
+                        k,
+                        v,
+                        seq_lens_info_q: SeqLensInfo,
+                        seq_lens_info_kv: SeqLensInfo,
+                        dropout_p=0.,
+                        sm_scale=None,
+                        attn_mask_type=None):
```

Comment:
It may be more clear if we check attn mask type in ColoAttention then pass `pad` and `causal` to this interface

Reply:
> It may be more clear if we check attn mask type in ColoAttention then pass `pad` and `causal` to this interface

updated.